src/components/QueryEditor/OpenAIEditor.tsx

```diff
@@ -47,9 +53,17 @@ export const OpenAIEditor: React.FC<RawQueryEditorProps> = (props) => {
   const [generatedQuery, setGeneratedQuery] = useState('//OpenAI generated query');
   const [variables] = useState(getTemplateSrv().getVariables());
   const [stateSchema, setStateSchema] = useState(cloneDeep(schema));
+  const [showQueryHistory, setShowQueryHistory] = useState(false);
+  const [parsedStoredPrompts, setParsedStoredPrompts] = useState([]);
   const styles = useStyles2(getStyles);
   const baselinePrompt = `You are an AI assistant that is fluent in KQL for querying Azure Data Explorer and you only respond with the correct KQL code snippets and no explanations. Generate a query that fulfills the following text.\nText:"""`;
 
+  useEffect(() => {
```

Comment:
This effect will run on every re-render, are we sure we want that to be the case?

Reply:
I think an empty `[]` will only run on "mount", if you omit the deps entirely that will run every re-render.